forest/shared/src/cli/snapshot_fetch.rs

```diff
@@ -12,39 +13,132 @@ use forest_utils::{
 };
 use hex::{FromHex, ToHex};
 use log::info;
+use regex::Regex;
 use s3::Bucket;
 use sha2::{Digest, Sha256};
+use std::str::FromStr;
 use std::{
     path::{Path, PathBuf},
     process::Command,
     time::Duration,
 };
+use time::{format_description, format_description::well_known::Iso8601, Date};
 use tokio::{
     fs::{create_dir_all, File},
     io::{AsyncWriteExt, BufWriter},
 };
 use url::Url;
 
-use crate::cli::to_size_string;
+/// Snapshot fetch service provider
+#[derive(Debug)]
+pub enum SnapshotServer {
+    Forest,
+    Filecoin,
+}
+
+impl FromStr for SnapshotServer {
+    type Err = anyhow::Error;
+
+    fn from_str(provider: &str) -> Result<Self, Self::Err> {
+        match provider.to_lowercase().as_str() {
+            "forest" => Ok(SnapshotServer::Forest),
+            "filecoin" => Ok(SnapshotServer::Filecoin),
+            _ => bail!(
+                "Failed to fetch snapshot from: {}, Must be one of `forest`|`filecoin`.",
+                provider
+            ),
+        }
+    }
+}
+
+/// Snapshot attributes
+pub struct SnapshotInfo {
+    pub network: String,
+    pub date: Date,
+    pub height: i64,
+    pub path: PathBuf,
+}
+
+impl SnapshotInfo {
+    fn new(path: PathBuf) -> Self {
+        Self {
+            network: "forest".into(),
+            date: Date::MIN,
+            height: 0,
+            path,
+        }
+    }
+}
+
+/// Collection of snapshots
+pub struct SnapshotStore {
+    pub snapshots: Vec<SnapshotInfo>,
+}
+
+impl SnapshotStore {
+    pub fn get_snapshots(
+        config: &Config,
+        snapshot_dir: Option<PathBuf>,
+    ) -> anyhow::Result<SnapshotStore> {
```

Comment:
As you've written it, this function cannot return anything other than `Ok`.

```suggestion
    ) -> SnapshotStore {
```

Reply:
fixed